patches/apply-patches.js

```diff
@@ -0,0 +1,92 @@
+/**
```

Comment:
@pdcastro since this file isn't run through the ts transpilation step it needs to be directly compatible with node 10 rather than being able to rely on transpilation - is it definitely compatible?

Reply:
Good point! 👍   I had tested with Node 10.21.0 and it was OK, but thanks to your question I tested it with 10.0.0 and found (as expected) that `const { promises: fs } = require('fs');` was not OK. It is best to have it compatible with 10.0.0 if easy to achieve (even if `engines.node` says `>=10.20.0`) because this allows more graceful error / warning message to be printed when users attempt to run the CLI. So I have made some changes and `apply-patches.js` is now compatible with Node v10.0.0.

Reply:
Did it fail with 10.0.0 @pdcastro or just throw the experimental warning?

Reply:
iirc, it was proper fatal failure, not just a warning.

Reply:
That's really weird, the docs say they were all added as part of 10.0.0

Reply:
The Changelog for Node.js mentions that `fs/promises` was moved to `fs.promises` in v10.1.0, and I've found the following comment:
> https://github.com/jsdom/jsdom/issues/2939#issuecomment-612193437
> _It does appear that anything `<10.1` will fail because `fs.promises` doesn't exist until then. Before `10.1` it was `fs/promises`._

Maybe this is what I was coming across when the script had failed.